scrape/scrape.go

```diff
@@ -1027,8 +1029,15 @@ mainLoop:
 func (sl *scrapeLoop) scrapeAndReport(interval, timeout time.Duration, last time.Time, errc chan<- error) time.Time {
 	start := time.Now()
 
-	// Only record after the first scrape.
 	if !last.IsZero() {
+		// Apply some tolerance on the scrape timestamp to align the samples
+		// correctly between scrapes.
+		exactStart := last.Add(interval)
+		if t := start.Sub(exactStart); t >= -scrapeTimestampTolerance && t <= scrapeTimestampTolerance {
+			start = exactStart
+		}
+
+		// Only record after the first scrape.
```

Comment:
I don't think this comment should move - it applies to the `IsZero()` check.

Reply:
Yeah. Is this code being added in the right place?

Reply:
I have moved it to the dedicated block, but it is the right place to me.